src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SQLDNSCache.cs

```diff
@@ -0,0 +1,87 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Concurrent;
+
+namespace Microsoft.Data.SqlClient
+{
+    internal class SQLDNSCache
+    {
+        private static readonly SQLDNSCache _SQLDNSCache = new SQLDNSCache();
+        private static readonly int initialCapacity = 100;
```

Comment:
What happens if capacity is exhausted?

Reply:
Good question. According to the [document](https://docs.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentdictionary-2.-ctor?view=netcore-3.1):

>The default capacity (DEFAULT_CAPACITY), which represents the initial number of buckets, is a trade-off between the size of a very small dictionary and the number of resizes when constructing a large dictionary. Also, the capacity should not be divisible by a small prime number. The default capacity is 31.

When the capacity is reached, it will resize the dictionary. I will change it to a prime number as the initial capacity, e.g. 101